astropy/modeling/models.py

```diff
@@ -6,15 +6,152 @@
 
 from . import math_functions as math
 from .core import custom_model, fix_inputs, hide_inverse
-from .functional_models import *
-from .mappings import *
-from .physical_models import *
-from .polynomial import *
-from .powerlaws import *
-from .projections import *
-from .rotations import *
-from .spline import *
-from .tabular import *
+from .functional_models import (
+    Scale,
+    Shift,
+    Multiply,
+    Planar2D,
+    Exponential1D,
+    Logarithmic1D,
+    RedshiftScaleFactor,
+    Ring2D,
+    AiryDisk2D,
+    Moffat1D,
+    Moffat2D,
+    Box1D,
+    Box2D,
+    Const1D,
+    Const2D,
+    Ellipse2D,
+    Disk2D,
+    Gaussian1D,
+    Gaussian2D,
+    Linear1D,
+    Lorentz1D,
+    RickerWavelet1D,
+    RickerWavelet2D,
+    Sersic1D,
+    Sersic2D,
+    Sine1D,
+    Cosine1D,
+    Tangent1D,
+    ArcSine1D,
+    ArcCosine1D,
+    ArcTangent1D,
+    Trapezoid1D,
+    TrapezoidDisk2D,
+    Voigt1D,
+    KingProjectedAnalytic1D,
+)
```

Comment:
@nstarman, you've been thinking more about the `__all__` API than me. Do you know of a more "elegant" solution to importing everything from the `__all__` of a submodule without using the `*`?

Reply:
> Do you know of a more "elegant" solution to importing everything from the __all__ of a submodule without using the *?

Well, either you import explicitly a long list of things (duplicating another long list in the imported module's `__all__`), with the risk that the two lists will not be consistent, or you use `import *`. 
This is one of the topics of the Public API discussion. 
(Scipy kept the `import *` after making the submodules private:
https://github.com/scipy/scipy/blob/main/scipy/ndimage/__init__.py#L152-L156)